### PR TITLE
Shadow3: add Shadow3 package

### PR DESCRIPTION
**Pull request by luccabavia**
Add Shadow3 package, a ray tracing modelling software used for x ray optics. 

@hyviquel

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-shadow3/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class PyShadow3(PythonPackage):
+    """SHADOW is an open source ray tracing code for modeling optical systems.
+    Targeted to synchrotron radiation beamlines, it has unique features for
+    designing X-ray optical systems.
+    """
+
+    homepage = "https://github.com/oasys-kit/shadow3"
+    url = "https://files.pythonhosted.org/packages/11/29/c330f8ed5fa1d77927105bb6884a928d74ac9ec89e03181f3084a6cdb361/shadow3-23.9.19-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-shadow3/package.py:16]: Can we use a .tar.gz file here? See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#id2

File: var/spack/repos/builtin/packages/py-shadow3/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    maintainers("luccabavia")
+
+    version("23.9.19", sha256=("d06920433552526398fd1373df03264a65dcdf020c0a67e196ac6f78f8673867"))
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-shadow3/package.py:22]: Missing build/run dep on numpy

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alecbcs** [comment]
Going to close this PR due to a lack of response after review feedback. @luccabavia happy to reopen if you still want to get this merged and have time to work on it.

**Pull request closed without merging**